### ISSUE TITLE
Visibility of accessor related to implicitly named creator affects delegating ctor selection

### ISSUE THREAD

**Issue by christophercurrie** [FasterXML/jackson-databind#870]
This may have to be part of the 2.7 rewrite, but I came across this trying to clean up for 2.6.

``` scala
case class Bar(value: String) // stand-in for a POJO.
case class Foo(bar: Bar) // in Java, a private "bar" field
 // and public "bar" getter. Implicit creator "bar".
mapper.readValue("{\"bar\":{\"value\":\"foo\"}}", classOf[Foo])
```

This deserializes using a property creator as expected. But if the `bar` property becomes "hidden":

``` scala
case class Bar(value: String)
case class Foo(private val bar: Bar) // in Java, obfuscated field and getter names.
  // Implicit creator still has the "bar" name from annotation introspector.
mapper.readValue("{\"bar\":{\"value\":\"foo\"}}", classOf[Foo])
```

This will fail to deserialize, because now Jackson thinks it should use a delegating creator, and attempts to deserialize the value using a `Bar` deserializer, which fails.

This might even "work as designed," but it's subtle and tripped me up trying to work out why the class wouldn't deserialize correctly.

**Comment by cowtowncoder** [comment]
Right, choice is certainly a mess, and I can not be 100% certain what works-as-designed would mean.

**Comment by cowtowncoder** [comment]
I would need a Java-only reproduction of this, to make sure if the problem still persists; and if so, if it might be easily solvable. I suspect both are possible given many fixes there have been (up to, say, 2.7.8) since this was filed.
